bool front
bool rear
bool front_toggle
bool rear_toggle